spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/sql/SqlServerJdbcDataSourcePropertiesUpdater.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.cloud.autoconfigure.sql;
+
+import com.microsoft.azure.management.sql.SqlServer;
+import com.microsoft.azure.spring.cloud.context.core.AzureAdmin;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+
+/**
+ * @author Warren Zhu
+ */
+public class SqlServerJdbcDataSourcePropertiesUpdater extends AbstractJdbcDatasourcePropertiesUpdater
+        implements JdbcDataSourcePropertiesUpdater {
+
+    public SqlServerJdbcDataSourcePropertiesUpdater(AzureSqlProperties azureSqlProperties, AzureAdmin azureAdmin) {
+        super(azureSqlProperties, DatabaseType.SQLSERVER, azureAdmin);
+    }
+
+    @Override
+    String getUserName() {
+        SqlServer sqlServer = azureAdmin.getSqlServer(azureSqlProperties.getServerName());
+        if (sqlServer == null) {
+            throw new IllegalArgumentException("SqlServer not found. If you want to auto create sqlServer. Please" +
+                    " provide username and password");
+        }
+
+        return sqlServer.administratorLogin();
```

Comment:
Not sure what this login is, any way we can get the username without login? If we have to login just in order to get the username, do we need to logout or handle the connection etc.?

Reply:
`administratorLogin` is actual the `username` we use to login the database